core/src/main/java/org/elasticsearch/discovery/zen/UnicastZenPing.java

```diff
@@ -508,7 +520,9 @@ public void run() {
                     }
                 });
             } else {
-                sendPingRequestToNode(sendPingsHandler.id(), timeout, pingRequest, latch, node, nodeToSend);
+                final DiscoveryNode finalNodeToSend = nodeToSend;
+                sendPingRequestToNode(() -> transportService.getConnection(finalNodeToSend),
```

Comment:
Question - why not just get the connection from the transport service right here? does the supplier buy as something? maybe we can even try to get it first and create a new temporary only if it fails
```
            Transport.Connection existingConnection = null;
            if (transportService.nodeConnected(nodeToSend)) {
                try {
                    existingConnection = transportService.getConnection(nodeToSend);
                } catch (NodeNotConnectedException e) {
                    // it's ok we'll create a temp connection
                }
            }

            if (existingConnection != null) {
                sendPingRequestToNode(existingConnection, sendPingsHandler.id(), timeout, pingRequest, latch, node, nodeToSend);
            } else {

```